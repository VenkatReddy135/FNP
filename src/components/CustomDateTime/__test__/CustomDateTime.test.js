import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import DateTimeInput from "../index";

jest.mock("react-admin", () => ({
  useInput: () => {},
  useTranslate: () => (label) => label,
}));

describe("CustomDateTime testcases", () => {
  beforeEach(() => {
    render(<DateTimeInput source="fromDate" label="From date" />);
  });

  test("test for render", () => {
    expect.assertions(1);
    expect(screen.getByTestId("CustomDateTimeInput")).toBeTruthy();
  });

  test("test for label", () => {
    expect.assertions(1);
    expect(screen.queryByText("From date")).toBeTruthy();
  });

  test("test for error on date select", () => {
    const date = screen.getByTestId("date");
    const time = screen.getByTestId("time");
    fireEvent.mouseDown(date);
    fireEvent.change(date, { target: { value: "2021-06-29" } });
    fireEvent.mouseDown(time);
    fireEvent.change(time, { target: { value: "11:50" } });
    expect(screen.getByDisplayValue("2021-06-29")).toBeTruthy();
    expect(screen.getByDisplayValue("11:50")).toBeTruthy();
  });

  test("tree for DateTimeInput", () => {
    const domTree = renderer.create(<DateTimeInput source="fromDate" label="From date" onChange={() => {}} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
