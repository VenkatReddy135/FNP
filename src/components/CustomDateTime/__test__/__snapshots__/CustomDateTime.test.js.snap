// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDateTime testcases tree for DateTimeInput 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-textInputWrap-242"
  data-testid="CustomDateTimeInput"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
    data-shrink={true}
  >
    From date
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
    onClick={[Function]}
  >
    <div
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-around"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="date"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-timePicker-191"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
              disabled={true}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="time"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8203;",
        }
      }
    />
  </p>
</div>
`;
